website/blog/2025-05-04-mlflow-image-datasets/index.md

```diff
@@ -0,0 +1,271 @@
+Furthermore, we will use the `mlflow.pytorch` [module doc](https://mlflow.org/docs/latest/api_reference/python_api/mlflow.pytorch.html#mlflow.pytorch.log_model) to log a model.
+
+## Example Using Computer Vision Model and Image Datasets Tracking
+
+There are two approaches to log an image dataset—using `mlflow.artifacts` or `mlflow.data` (datasets). Lastly, you can also log an evaluation dataset, which I will not cover in this post.
+
+Why are there two approaches? Because converting an image dataset from file-based formats like COCO to a tabular format is challenging, as most dataloaders expect the file-based COCO format. Logging artifacts provides a quick and straightforward solution without the need to reformat files, but it can easely also be messy. Make sure to create meaningful paths for your artifacts.
+
+The key artifact for COCO datasets is the `instances.json` file, which describes your image dataset's metadata and annotations. This file can, for example, be used to check the class balance in your dataset by analyzing the `category` field.
+
+If you are not too concerned about this, Hugging Face can help with logging datasets the MLflow way. Some Hugging Face datasets contain rich metadata that can be transferred to MLflow tracking. This is where `mlflow.data` comes in. Compared to logging artifacts, this add much more reach metadata and structure to your dataset and makes it easier to manage and see in the experiment run. If you can fit your dataset into a Huggingface type dataset and make that work in your dataloader or training script, this is the recommended apporach.
+
+In this post, I will go through both approaches in code.
+
+### Install MLflow and Other Dependencies
+
+Start by installing the dependencies for both code examples in your `python >= 3.10` environment. opencv can be ommited if only using the first examples and pycocotools can be ommitted if only using the second example.
+
+`pip install mlflow datasets torch torchvision pycocotools opencv-python-headless psutil`
+
+Also install `pynvml` if you want to track GPU metrics.
+
+For the Hugging Face dataset download, also make sure to log in.
+
+`huggingface-cli login`
+
+One of the examples requires compute; therefore, be sure to turn on [MLflow system metrics](https://mlflow.org/docs/latest/system-metrics/) to track what happens on your compute during training.
+
+`export MLFLOW_ENABLE_SYSTEM_METRICS_LOGGING=true`
+
+> Note: The validation split is used to save space, but you can also use the "train" split 
+> if you want to train/finetune the model on the full dataset (requires +25 GB storage).
+> Number of epochs and a subset of the dataset is also used during training.
+
+### Logging Artifacts of Datasets Together with Model
+Since the COCO dataset is filebased, files need to be downloaded first. We use the smallest version of the newest version of the dataset from the official author's website.
```

Comment:
```suggestion
Since the COCO dataset is file-based, files need to be downloaded first. We use the smallest version of the newest version of the dataset from the official author's website.
```